Adds support to specify a gradient_by cohort or pivot columns

MiguelMadero opened the pull request:
When defining the gradient based on the pivot_column we can more easily compare how well each cohort performed at any given point in time. cc: @leqilong

leqilong commented:
Hi @MiguelMadero , thank you for the PR. The code looks good and seems to be non-breaking from what I've seen. Just one small thing, you would also want to update the minified version of `alamode.js` in this PR. We have an instruction [here](https://github.com/mode/alamode#minification) on how to minify it. Once you update the minified file, I'll go ahead and approve. Thanks!

MiguelMadero commented:
@leqilong thanks! and sorry for the delay, I just saw this. I rebased from master and updated the minified file. 

Please let me know if there is anything else missing. I'll keep a closer eye on this repo.

leqilong commented:
Thanks @MiguelMadero ! I've deployed your changes!
